packages/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -46,3 +29,88 @@ export function toBigNumber(input: string) {
 export function toNumber(input: string) {
   return parseInt(input, 10)
 }
+
+export function parseNumber(input: string | number | BigNumber) {
+  return new BigNumber(input).toString()
+}
+
+type Parser<A, B> = (input: A) => B
+
+export function tupleParser<A0, B0>(parser0: Parser<A0, B0>): (...args: [A0]) => [B0]
```

Comment:
Feels like there should be a better way to do this function def eh? But maybe not supported

Reply:
if you have some typescript typing mastery and want to suggest something, please :D

This was the best solution that i could think of... and not an uncommon one.

The problem here is that an array of Parsers won't work, since each parse has a different type variables. I don't know how to type an arbitrary tuple of different parsers, and then define a function which maps each position parser input to the parse output.

Using function overloading was my best solution to that, and one that i've seen in typescript or even in other languages (haskell)